File: inform/Preferences/IFNewThemeWindow.m
//
//  IFNewThemeWindow.m
//  Inform
//
//  Created by Toby Nelson on 15/05/2022.
//

#import <Foundation/Foundation.h>
#import "IFNewThemeWindow.h"

@implementation IFNewThemeWindow {
    IBOutlet NSButton*      okButton;
    IBOutlet NSButton*      cancelButton;
    IBOutlet NSTextField*   nameField;

    NSString *              _themeName;
}

- (instancetype) init {
    self = [super init];

    if (self) {
    }
    return self;
}

-(NSString *) themeName {
    return _themeName;
}

-(void) setThemeName:(NSString *)theName {
    _themeName = theName;
    [nameField setStringValue: theName];
}

-(IBAction) okButtonClicked:(id) sender {
    self.themeName = [nameField stringValue];
    [NSApp endSheet:self returnCode: NSModalResponseOK];
}

-(IBAction) cancelButtonClicked:(id) sender {
    self.themeName = @"";
    [NSApp endSheet: self returnCode: NSModalResponseAbort];
}

@end

File: inform/Preferences/IFNewThemeWindow.swift
//
//  IFNewThemeWindow.swift
//  Inform
//
//  Created by Toby Nelson on 28/09/2023.
//

import Foundation

class IFNewThemeWindow: NSWindow {
    @IBOutlet var okButton: NSButton!
    @IBOutlet var cancelButton: NSButton!
    @IBOutlet var nameField: NSTextField!
    @objc public var themeName: String!

    func setThemeName(theName:String!) {
        themeName = theName
        nameField.stringValue = theName
    }

    @IBAction func okButtonClicked(sender:AnyObject!) {
        themeName = nameField.stringValue
        NSApp.endSheet(self, returnCode: NSApplication.ModalResponse.OK.rawValue)
    }

    @IBAction func cancelButtonClicked(sender:AnyObject!) {
        themeName = ""
        NSApp.endSheet(self, returnCode: NSApplication.ModalResponse.abort.rawValue)
    }
}
